### PR TITLE
Tool to ingest ZTF source features

### PR DESCRIPTION
**Pull request by dmitryduev**
This PR implements a tool to ingest ZTF source features produced by @mcoughlin + external features (from cross-matches) + dmdt's.

### CODE REVIEW

File: kowalski/utils.py [hunk 1]
```diff
@@ -1014,14 +1014,22 @@ def desi_dr8_url(ra, dec):
 
 
 @jit
-def pwd_for(a):
+def pwd_for(a: Sequence):
     """
     Compute pairwise differences with for loops
     """
     return np.array([a[j] - a[i] for i in range(len(a)) for j in range(i + 1, len(a))])
```
Comment (mcoughlin) [inline kowalski/utils.py:1021]: I guess this is faster than np.diff and friends?

**Review by mcoughlin — APPROVED** [review]
Looks good to me.

File: kowalski/utils.py (hunk 1 above)
Comment (dmitryduev) [reply kowalski/utils.py:1021]: yes, the jit-compiled version is indeed faster on average.

**Review by mcoughlin — APPROVED** [review]

**Review by kmshin1397 — COMMENTED** [review]
Couple of minor things, but otherwise looks good! Nothing stands out to me as obvious opportunities for optimization either. 

Also, probably not for this PR, but thoughts on tweaking the configs/tests so that the tests run against its own test database like in SkyPortal? Or just special test collections i.e. `test_ZTF_alerts` instead of having the ingester tests post directly to `ZTF_alerts`? I guess since K tests run with what I think are real data samples it's not as important to isolate them?

File: tools/ingest_ztf_source_features.py [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+        """ catalogs """
+        catalog = "ZTF_alerts"
+        object_position_query = dict()
+        object_position_query["coordinates.radec_geojson"] = {
+            "$geoWithin": {
+                "$centerSphere": [[ra_geojson, dec_geojson], CONE_SEARCH_RADIUS]
+            }
+        }
+        # n = int(_db[catalog].count_documents(object_position_query))
```
[the remark this answers is not in the document]
Comment (dmitryduev) [reply tools/ingest_ztf_source_features.py:190]: Rm'ed

File: tools/ingest_ztf_source_features.py [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+                "$nearSphere": [ra_geojson, dec_geojson],
+                "$minDistance": 0,
+                "$maxDistance": CONE_SEARCH_RADIUS,
+            }
+            s = _mongo.db[catalog].find(
+                {**object_position_query, **catalog_filter},
+                {**catalog_projection},
+                limit=1,
+            )
+            cross_matches[catalog] = list(s)
```
[the remark this answers is not in the document]
Comment (dmitryduev) [reply tools/ingest_ztf_source_features.py:121]: `find` returns a Cursor, feeding that into `dict(...)` will err, I think. One could do `find_one`, which returns a `dict`, but it returns the whole document, you can't apply projections - so we'll get a lot of unneeded data. I'd leave it as is.

**Comment by dmitryduev** [comment]
Failure on the CI is unrelated (something's up with the TNS), so should be mergeable @kmshin1397.

File: tools/ingest_ztf_source_features.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kmshin1397) [reply tools/ingest_ztf_source_features.py:121]: Ah ok got ya, I missed that find returns a cursor. That makes sense then.

**Pull request merged**